### ISSUE TITLE
Badgecount when app is closed

### ISSUE THREAD

**Issue by Ongatkha** [evollu/react-native-fcm#549]
The message has both notif and data payload.
Any handler?

**Comment by evollu** [comment]
badge count should be in notif payload

**Comment by Ongatkha** [comment]
@evollu how about doing it native? on android i can through the onIntent or whatever its called now handler.

**Comment by evollu** [comment]
sorry I don't get your question

**Comment by Ongatkha** [comment]
@evollu on android you have a method that gets called wether the notification was called in background/foreground or closed that i can use to update my badgecount number
public void onNewIntent(Intent intent){}

Is there such thing on ios?

**Comment by evollu** [comment]
yes. you just set "badge" property in notification payload.

**Comment by Ongatkha** [comment]
@evollu yes but without setting it on the server side, i keep it at 1 and i modified the code to natively keep count whenever i receive one.

**Comment by evollu** [comment]
it is not reliable because when iOS app is killed by user no code will be run when notif arrives

**Comment by vvkaghera** [comment]
@evollu  _it is not reliable because when iOS app is killed by user no code will be run when notif arrives.
if you don't care you can call FCM.setBadgeCount(x)_

Where can I add code ` FCM.setBadgeCount(x)`

**Comment by evollu** [comment]
@vvkaghera you can add it in notification callback
